app/src/main/java/me/guillaumin/android/osmtracker/db/DataHelper.java

```diff
@@ -65,6 +68,9 @@
 	 */
 	private Context context;
 
+	/** Statistics for all existing tracks */
+	private static TreeMap<Long, TrackStatistics> tracksStatistics = new TreeMap<Long, TrackStatistics> ();
```

Comment:
>  wouldn't I have to write some additional complicated code to figure out which trackpoints aren't a part of the statistics yet?

No I wouldn't, just add a WHERE clause to the database query, so that's out of the way.